### PR TITLE
Package reorganization (step 2)

### LINKED ISSUE #2525: [Feature Request] Reorganize package names and redistribute some styles, especially from USER-MISC
**Summary**

There is a need to reorganize the location of some styles for consistency and better maintainability.
Especially the USER-MISC package is getting too big and we should consider creating, e.g. a USER-MANYBODY and/or a USER-MOLECULE and/or a USER-KSPACE package. Similarly, some of the less commonly used styles in src/ could be moved to MISC (or USER-MISC?).

When discussing with contributors it is getting sometimes difficult to justify why they are asked to do things differently than it was for other contributions.

When shuffling things around (and thus confusing some users), we should also consider renaming a few more packages with respect to their status as "user" package (to have the confusion only once and not twice or more often). USER-COLVARS, USER-REAXC, USER-MEAMC, and possibly also USER-OMP could lose the USER prefix and LATTE, MSCG, POEMS, and MPIIO look like it would be justified to add a USER prefix when looking at the maintenance status and type of contribution. I have already multiple times voiced my concern that the designation of user packages as given in the relevant documentation and web pages does not match well with the current reality.

----

After offline discussions with @sjplimp the proposal is now the following:
- we rename some packages that have their names for historical reasons (e.g. USER-MEAMC to MEAM, USER-REAXC to REAX or REAXFF), also file names, class names and style name may be renamed (with backward compatibility, as already available in meam/c)
- we review the contents of USER-MISC and move styles that comply with strict coding standards of the core LAMMPS code to existing packages (e.g. MOLECULE, KSPACE, MANYBODY, MISC) where applicable. 
- less commonly used styles in src are moved to appropriate packages (e.g. MISC)
- we create a few new packages to group thematically styles that don't fit well elsewhere and otherwise would stay in USER or go to MISC. that could be thermostat/time-integrator fixes, pair styles, computes/fixes for analysis.
- new contributions from users that are just a single style are initially added to the USER-MISC package in the same way as is done with USER-MISC right now. depending on the details of the implementation and the application another package may be suggested.
- the conditions for creating a new package are either 1) 3 or more styles that are thematically closely related where the purpose does not match well with existing packages or 2) code that depends on an external or bundled library or use a different programming language (e.g. Fortran, CUDA, etc.)

----

#### Renamed packages
- MLIAP -> **ML-IAP**
- SNAP -> **ML-SNAP**
- USER-ADIOS -> ADIOS
- USER-ATC -> ATC 
- USER-AWPMD -> AWPMD
- USER-BOCS -> BOCS
- USER-BROWNIAN -> BROWNIAN
- USER-CGDNA -> **CG-DNA**
- USER-CGSDK -> **CG-SDK**
- USER-COLVARS -> COLVARS
- USER-DIELECTRIC -> DIELECTRIC
- USER-DIFFRACTION -> DIFFRACTION
- USER-DPD -> **DPD-REACT**
- USER-DRUDE -> DRUDE
- USER-EFF -> EFF
- USER-FEP -> FEP
- USER-H5MD -> H5MD
- USER-HDNNP -> **ML-HDNNP**
- USER-INTEL -> INTEL
- USER-LB -> **LATBOLTZ**
- USER-MANIFOLD -> MANIFOLD
- USER-MEAMC -> **MEAM**
- USER-MESODPD -> **DPD-MESO**
- USER-MDI -> MDI
- USER-MESONT -> MESONT
- USER-MGPT -> MGPT
- USER-MISC -> USER-MISC
- USER-MOFF -> MOFF
- USER-MOLFILE -> MOLFILE
- USER-NETCDF -> NETCDF
- USER-OMP -> **OPENMP**
- USER-PACE -> **ML-PACE**
- USER-PHONON -> PHONON
- USER-PLUMED -> PLUMED
- USER-PTM -> PTM
- USER-QMMM -> QMMM
- USER-QTB -> QTB
- USER-QUIP -> **ML-QUIP**
- USER-RANN -> **ML-RANN**
- USER-REACTION -> REACTION
- USER-REAXC -> **REAXFF**
- USER-SCAFACOS -> SCAFACOS
- USER-SDPD -> **DPD-SMOOTH**
- USER-SMD -> **MACHDYN**
- USER-SMTBQ -> SMTBQ
- USER-SPH -> **SPH**
- USER-TALLY -> TALLY
- USER-UEF -> UEF
- USER-VTK -> VTK
- USER-YAFF -> YAFF

#### Possible new packages
- ANALYSIS for less commonly use computes and related fixes like fix ave/correlate/long
- INTERLAYER (or ILP?) pair styles: drip, ilp/graphene/hbn, kolmogorov/crespi/full, kolmogorov/crespi/z, lebedeva/z
- EXTRA-PAIR for less commonly used pair styles, and pair styles associated with large(r) potential files so that the potential files are no longer bundled, but will be downloaded on the fly, if not already present, during package installation. cf. USER-MESONT

The full list of packages would then become something like (plus any new packages):

ADIOS ASPHERE ATC AWPMD BOCS BODY CGDNA CGSDK CLASS2 COLLOID COLVARS COMPRESS CORESHELL DIFFRACTION DIPOLE DPD-MESO DPD-REACT DPD-SMOOTH DRUDE EFF FEP GPU GRANULAR H5MD INTEL INTERLAYER KIM KOKKOS KSPACE LATTE LATBOLTZ MACHDYN MANIFOLD MANYBODY MC MEAM MESONT MESSAGE MGPT MISC ML-IAP ML-PACE ML-QUIP ML-SNAP MOFFF MOLECULE MOLFILE MPIIO MSCG NETCDF OPENMP OPT PERI PHONON PLUMED POEMS PTM PYTHON QEQ QMMM QTB REACTION REAXFF REPLICA RIGID SCAFACOS SHOCK SMTBQ SPH SPIN SRD TALLY UEF USER-MISC VORONOI VTK YAFF 

### Examples tree

The examples tree will remain mostly the same (for now). The examples/USER folder will be renamed to examples/PACKAGE

### PR DESCRIPTION
**Pull request by akohlmey**
**Summary**

This pull request continues the reorganization of packages mainly by moving styles away from USER-MISC to existing packages or by creating new packages.

**Related Issue(s)**

continues from #2818 
closes #2525 

**Author(s)**

Axel Kohlmeyer, Temple U

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

No functional changes are included, but configuration and compilation steps may change due to changed/new package names.

**Implementation Notes**

The following individual sets of changes are included:
- move inter-layer potentials from USER-MISC to a new package INTERLAYER, examples are in examples/PACKAGES/interlayer
- move bond, angle, dihedral, improper potentials (21) from USER-MISC to MOLECULE package (except for bond style special)
- move bond style special, pair style list pair style srp, fix addtorque, fix imd, fix gle, fix smd from USER-MISC to MISC package
- move temper_grem and temper_npt commands and related internal styles from USER-MISC to REPLICA package
- move angle style dipole, pair style dipole/sf, pair style  from USER-MISC to DIPOLE package

**Post Submission Checklist**

- [x] The feature or features in this pull request is complete
- [x] Licensing information is complete
- [x] Corresponding author information is complete
- [x] The source code follows the LAMMPS formatting guidelines
- [x] Suitable new documentation files and/or updates to the existing docs are included
- [x] The added/updated documentation is integrated and tested with the documentation build system
- [x] The feature has been verified to work with the conventional build system
- [x] The feature has been verified to work with the CMake based build system

### CODE REVIEW

File: src/Makefile [hunk 1]
```diff
@@ -186,7 +188,7 @@ PACKMOST = \
 	ml-iap \
 	phonon
 
-# NOTE: the last three packages must remain at the end since
+# NOTE ^^^^^: the last three packages must remain at the end since
```
Comment (rbberger) [inline src/Makefile:191]: @akohlmey typo?

Comment (akohlmey) [reply src/Makefile:191]: No, this is intentional. Comments are usually before relevant sections, but this is after because the list is so long. So I thought a little visual hint might be helpful.

**Comment by akohlmey** [comment]
@sjplimp @athomps @stanmoore1 @rbberger 
This has now all creation of new packages done and also moving of source files around done based on the discussions between @sjplimp and me.
I still need to do some updates and testing with the dependency tracking for the GNU make build process and need to update the manual and the individual doc pages of the affected commands (and write some scripts for checking this). also, I would need to move some more example folders around as well.

If you see something that is moving a style into the wrong

Bottom line:
- USER-MISC is gone
- MISC has a small number of rather unusual styles left and pair style agni (which is earmarked for an ML-MISC or ML-EXTRA package as soon as we have some additional machine learning potentials that don't need a package of their own).
 @athomps how do you feel about adding agni to ML-IAP? I don't think it is a great fit, so I stuck it into MISC instead.
- there are EXTRA-PAIR, EXTRA-COMPUTE, EXTRA-FIX, EXTRA-MOLECULE with less commonly used styles from either USER-MISC, src, or MOLECULE

**Comment by athomps** [comment]
pair style agni would not make sense in ML-IAP, despite the name.  It can go in MISC for now.  I think ML-MISC is also a good option.

**Review by athomps — APPROVED** [review]
I approve.

**Review by sjplimp — APPROVED** [review]
